### DISCUSSION TITLE
one-line scripts vs plugins

### DISCUSSION THREAD

**Discussion by laur89** [VR-25/acc#230] [Q&A] [answered]
Readme states
>An alternative to plugins are one-line scripts. Refer to the [default configuration](https://github.com/VR-25/acc#default-configuration) section above.

The default config section has:
```// one-line scripts # Default: none

// Every line that begins with ":" is interpreted as a shell script.
// This feature can be useful for many things, including setting up persistent config profiles (source a file that overrides the main config).
```
---
What exactly is difference between `plugins` and `one-line scripts`? Aren't these one-line scripts lines that need to reside in a plugin .sh file that needs to be in `/plugins` dir?

**Comment by VR-25** [comment, marked as answer]
Plugins are actual sh files and can contain multiple lines.

One-line scripts are part of the config itself. This makes them simpler, easier to work with and more dynamic.

Plugins are usually meant for overriding internal functions and  variables.
Unlike one-line scripts, they are loaded only once, when the daemon starts.

Among many possibilities, one-line scripts can be used for scheduling.
I use these for overriding virtual memory parameters as well (which has nothing to do with acc):

`: vm; vm="swappiness=1 vfs_cache_pressure=10 dirty_expire_centisecs=0 dirty_background_ratio=2 dirty_writeback_centisecs=0 dirty_ratio=75"; for i in $vm; do sysctl -w vm.$i >/dev/null; done`
